tests/components/group/test_init.py

```diff
@@ -659,6 +659,24 @@ async def test_is_on(hass: HomeAssistant) -> None:
             (STATE_ON, True),
             (STATE_OFF, False),
         ),
+        (
+            ("vacuum", "vacuum"),
+            # Cleaning is the only on state
+            (vacuum.STATE_DOCKED, vacuum.STATE_CLEANING),
+            # Returning is the only on state
+            (vacuum.STATE_RETURNING, vacuum.STATE_PAUSED),
+            (vacuum.STATE_CLEANING, True),
+            (vacuum.STATE_RETURNING, True),
+        ),
+        (
+            ("vacuum", "vacuum"),
+            # Multiple on states, so group state will be STATE_ON
+            (vacuum.STATE_RETURNING, vacuum.STATE_CLEANING),
+            # Only off states, so group state will be off
+            (vacuum.STATE_PAUSED, vacuum.STATE_IDLE),
+            (STATE_ON, True),
```

Comment:
As written on discord to @jbouwh I don't like this mixin of a generic group state with platform dependent states but probably that's the way this needs to go?